### ISSUE TITLE
Autoformatting

### ISSUE THREAD

**Issue by ivov** [MLH-Fellowship/n8n#37]
With Prettier we can auto format almost everything according to codebase conventions. But Prettier also enforces _auto collapsing_ for arrays, object key value pairs and imports, all based on whether they exceed print width. This auto collapsing, as far as I know, [cannot be disabled](https://github.com/prettier/prettier-vscode/issues/352). I looked through [the Prettier source](https://github.com/prettier/prettier/tree/master/src/language-js) but couldn't find how to stop it.

If n8n were willing to relax the rules on multiline formatting, then Prettier allows for auto insertion of trailing commas and semicolons and auto conversion of quotes, which now has to be done manually and then also reviewed manually by us and the n8n team. `.editorconfig` does not enforce trailing commas, semicolons and quotes. Another option would be to use Prettier for auto formatting, and then turn it off and manually recreate the multiline format, but the end result would be the same - just one kind of manual formatting instead of another.

Also, TSLint is being deprecated in favor of ESLint. While ESLint is just a linter, you can use the `--fix` flag for auto formatting. But, ESLint support for multiline formatting is poor. ESLint offers the `array-element-newline` and `object-curly-newline` rules and the `modules-newline` plugin. Detection is great but the auto formatting is very clumsy. And it also affects lines like `if(['add', 'delete', 'update'].includes(operation))` where we actually want to keep this in a single line.

```js
// by hand
operation: [
  'add',
  'delete',
]

// Prettier
operation: [ 'add', 'delete', ]

// ESLint
operation: [ 'add', 
'delete', ]
```

So in short, it would be great to have auto formatting to make it easier for everyone, but on the other hand the multiline style is very prevalent throughout the codebase. So it's a big tradeoff.
